### PR TITLE
feat: added rsv modules

### PR DESCRIPTION
**Pull request by alexanderojala**
## Description

If you haven't already, read the full [contribution guide](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

Enable GitHub Workflows in your fork to enable auto-generation of assets with our [GitHub Action](/.github/workflows/push-auto-generate.yml).
To trigger GitHub Actions after auto-generation, [add a GitHub PAT](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) as a secret in your forked repository called `PAT`.

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [ ] I have updated the examples in README with the latest module version number.

### CODE REVIEW

**Review by alexanderojala — COMMENTED** [review]
dd

File: avm/res/recovery-services/vault/replication-policy/version.json
Comment (eriqua) [inline avm/res/recovery-services/vault/replication-policy/version.json:1]: Please remove version.json files from all child modules. Differently than CARML, the current CI is not publishing child modules separately.

File: avm/res/recovery-services/vault/version.json
Comment (eriqua) [inline avm/res/recovery-services/vault/version.json:3]: Please start the main version from 0.1
```suggestion
    "version": "0.1",
```

**Review by eriqua — CHANGES_REQUESTED** [review]
Hey @alexanderojala thanks a lot for this contribution and for owning this module!
Before initial code review, please make sure the following is also added:

- codeowners file should be updated as per the following guidelines https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships
- update issue template as per https://azure.github.io/Azure-Verified-Modules/specs/bicep/#id-bcpnfr15---category-contributionsupport---avm-module-issue-template-file
- add corresponding github workflow
- attach a pipeline badge as a proof that the module is validated in you environment

You can reference PR #874 as an example for above requested additions

**Comment by ChrisSidebotham** [comment]
@alexanderojala @ms-ojala - Have you been able to perform any of the following:


- codeowners file should be updated as per the following guidelines https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships
- update issue template as per https://azure.github.io/Azure-Verified-Modules/specs/bicep/#id-bcpnfr15---category-contributionsupport---avm-module-issue-template-file
- add corresponding github workflow
- attach a pipeline badge as a proof that the module is validated in you environment
- You can reference PR https://github.com/Azure/bicep-registry-modules/pull/874 as an example for above requested additions

**Comment by alexanderojala** [comment]
Kindly ignore this PR, was faulty. I will create a new PR once module ready for merge

**Pull request closed without merging**
